sdk/keyvault/keyvault-keys/src/lro/keyVaultKeyPoller.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { delay, RequestOptionsBase } from "@azure/core-http";
+import { Poller, PollOperation, PollOperationState } from "@azure/core-lro";
+import { KeyVaultClient } from "../generated/keyVaultClient";
+
+/**
+ * Common parameters to a Key Vault Key Poller.
+ */
+export interface KeyVaultKeyPollerOptions {
+  vaultUrl: string;
+  client: KeyVaultClient;
+  name: string;
+  requestOptions?: RequestOptionsBase;
+  intervalInMs?: number;
+  resumeFrom?: string;
+}
+
+/**
+ * An interface representing the state of a Key Vault Key Poller's operation.
+ */
+export interface KeyVaultKeyPollOperationState<TResult> extends PollOperationState<TResult> {
+  /**
+   * The name of the key.
+   */
+  name: string;
+}
+
+/**
+ * Common properties and methods of the Key Vault Key Pollers.
+ */
+export abstract class KeyVaultKeyPoller<TState, TResult> extends Poller<TState, TResult> {
+  /**
+   * Defines how much time the poller is going to wait before making a new request to the service.
+   * @memberof DeleteKeyPoller
+   */
+  public intervalInMs: number = 2000;
+
+  /**
+   * The method used by the poller to wait before attempting to update its operation.
+   * @memberof DeleteKeyPoller
+   */
+  async delay(): Promise<void> {
+    return delay(this.intervalInMs);
+  }
+}
+
+/**
+ * Common properties and methods of the Key Vault Key Poller operations.
+ */
+export class KeyVaultKeyPollOperation<TState, TResult> implements PollOperation<TState, TResult> {
+  constructor(public state: TState, private cancelMessage: string) {}
```

Comment:
since cancelMessage feels a little niche, should we proactively make this an options bag that contains cancelMessage?